delivery_carrier_label_postlogistics/models/stock.py

```diff
@@ -24,6 +24,24 @@ class StockPicking(models.Model):
         "Mobile", help="For notify delivery by telephone (ZAW3213)"
     )
 
+    def _compute_show_label_button(self):
+        """Determine if we should show the button to generate labels
+
+        Allows to print the label early, as for return pickings when
+        the label must be printed and sent to the customer.
+
+        """
+        visible_states = ['confirmed', 'assigned', 'done']
+        postlog_pickings = self.filetered(
+            lambda rec: rec.carrier_id.delivery_type == 'postlogistics'
+        )
+        for pick in postlog_pickings:
+            pick.show_label_button = (
+                pick.state in visible_states and pick.carrier_code
```

Comment:
I'm just not sure about this, shouldn't the button be displayed only?
 - When state = done on outgoing pickings
 - When state = confirmed or assigned on incoming pickings